### ISSUE TITLE
Fix compilation warning on MacOS

### ISSUE THREAD

**Issue by plajjan** [actonlang/acton#295]
When compiling certain programs, like `test/stdlib/test_numpy.act`, a warning is emitted:
```
warning: (x86_64) /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/numpy-1a8b4b.o unable to open object file: No such file or directory
```

As far as I can tell, it has to do with generating debug symbols. We do ask cc to generate debug symbols with -g, but we have done that before. This warning started showing up with #293, which brought more consistency to how we apply `-g`. It also did do relocatable object files (for numpy.o at least), is that related?

Guess this is some yak shaving for a rainy day.

**Comment by plajjan** [comment]
http://wiki.dwarfstd.org/index.php?title=Apple%27s_%22Lazy%22_DWARF_Scheme

**Comment by plajjan** [comment]
Reclassifying. I think this has to do with dsym stuff on macos, no?

**Comment by plajjan** [comment]
This has to do with using -r for partially linking numpy.o with its dependencies (math.o) at build time of numpy.o. However, that was a broken solution since then could not import both numpy and math at the same time. I finally fixed that in #693, which also fixes this issue. There are no more warnings emitted! Closing...
